Framework/CurveFitting/src/Algorithms/PlotPeakByLogValue.cpp

```diff
@@ -392,7 +392,7 @@ void PlotPeakByLogValue::appendTableRow(
       row << ifun->getParameter(iPar) << ifun->getError(iPar);
     }
 
-    if (ifun->hasParameter("Intensity") == false) {
+    if (ifun->hasParameter("Intensity") == false && ifun->hasParameter("I") == false ) {
```

Comment:
I have to say the more I look at this, the uglier it looks.
Could we have a method `bool IPeakFunction.hasIntensityPamarater() const` so that we could write
```c++
bool IPeakFunction::hasIntensityParameter() {
return hasParameter("Intensity") || hasParameter("I");
}
```
and
```c++
if (intensity_handle and !intensity_handle->hasIntensityParameter()){
....
}

Reply:
Does IPeakFunction have method intensity()?  
Then this method will be overridden for each specific peak function regardless intensity is a native function parameter or an effective parameter.

Reply:
> Does IPeakFunction have method intensity()?
> Then this method will be overridden for each specific peak function regardless intensity is a native function parameter or an effective parameter.

@wdzhou Yes, intensity() is a virtual method in IPeakFunction. However, it does not need to be invoked if the intensity is a fitting parameter since the fitting parameters are automatically displayed. We could invoke it every time, and this would not produce incorrect results. It may just include 2 extra copied columns.

Reply:
> I have to say the more I look at this, the uglier it looks.
> Could we have a method `bool IPeakFunction.hasIntensityPamarater() const` so that we could write
@jmborr I moved that checking functionality into a dedicated function

Reply:
@jmborr introduced a bug, I'll have to finish this later

Reply:
@stonecoldhughes  After today's conversation at the Diffraction meeting, I don't think this is necessary since Yuanpeng doesn't mind having two columns with the same information. The conditional
```c++
if (ifun->hasParameter("Intensity") == false && ifun->hasParameter("I") == false )
```
seems a moot point now.